### DISCUSSION TITLE
JS | GraphiQL monorepo

### DISCUSSION THREAD

**Discussion by klippx** [renovatebot/renovate#33135] [Suggest an Idea]
### Tell us more.

Can we create a monorepo for GraphiQL?

- https://github.com/graphql/graphiql/blob/main/packages/graphiql/package.json#L2
- https://github.com/graphql/graphiql/blob/main/packages/graphiql-react/package.json#L2
- https://github.com/graphql/graphiql/blob/main/packages/graphiql-toolkit/package.json#L2
- etc...

Today we are getting individual PR's which are individuall breaking and we have to manually bundle / merge them into each other which is a pain.

<img width="754" alt="Screenshot 2024-12-16 at 9 44 52 AM" src="https://github.com/user-attachments/assets/0c259331-273c-4876-860c-e6647caee587" />

Is there anything I can do to help create a PR for creating a renovate monorepo preset?

**Comment by secustor** [comment]
Contributions welcome, you need to add it here: https://github.com/renovatebot/renovate/blob/main/lib/data/monorepo.json

**Reply by klippx** [reply]
Opened a PR https://github.com/renovatebot/renovate/pull/33188

**Reply by klippx** [reply]
Working great now!
<img width="582" alt="Screenshot 2024-12-19 at 10 03 04 AM" src="https://github.com/user-attachments/assets/78d54fe5-62c0-4bbd-bfef-d8dd9230b34e" />
<img width="641" alt="Screenshot 2024-12-19 at 10 02 52 AM" src="https://github.com/user-attachments/assets/762b1d29-58b4-41f8-a485-389f4759bd7d" />

**Comment by rarkins** [comment]
They are versioned differently, which makes things a bit annoying (creates immortal PRs)

**Reply by klippx** [reply]
You mean that `graphiql` would be better off if they kept versioning in sync as in both are `3.8.1` ?
Or is it enough if `@graphiql/react` in this case would not be in `0.x` but at least in `1.x` (stable) ?

**Reply by rarkins** [reply]
If Renovate has a group of dependencies, and they're all the same version, then the PR will be like "Update graphiql to v1.2.3". And you can close that PR to ignore v1.2.3. You'll get another when v1.2.4 or v1.3.0 comes along. However if versions are mixed, the PR will be "Update graphiql" and you can't close it to ignore it.

**Reply by klippx** [reply]
<img width="1023" alt="Screenshot 2024-12-18 at 12 52 19 PM" src="https://github.com/user-attachments/assets/9573ee64-0680-41f4-91e9-299246555f96" />

Isn't `opentelemetry` monorepo using a similar versioning strategy? To me it works "good enough", WDYT?

**Reply by rarkins** [reply]
Yes, but it's still not desirable, so should only be done if it's usually necessary to group them

**Reply by klippx** [reply]
Yea, it is necessary. The `graphiql` branch fails because it needs the new version of `@graphiql/react`. 

It is also not desirable to have multiple PRs for monorepos (less PRs = better). Especially when they are individually failing. I think that trumps the mild inconvenience of immortal PRs and PRs without version in title (even though I hear you loud and clear that this is a valid concern).

However I do not think that the maintainers of `graphiql` would be willing to change versioning strategy to satisfy renovate, and I am not sure I am prepared to ask them to do such a thing.

**Reply by klippx** [reply]
After discussions, I decided on opening a PR https://github.com/renovatebot/renovate/pull/33188
